salt/metalk8s/salt/master/certs/salt-api.sls

```diff
@@ -0,0 +1,46 @@
+{%- from "metalk8s/map.jinja" import kube_api with context %}
+
+include:
+  - metalk8s.internal.m2crypto
+
+Create Salt API private key:
+  x509.private_key_managed:
+    - name: /etc/kubernetes/pki/salt-api.key
```

Comment:
Salt and K8s are quite two different worlds. Should not we have a different path for both? Like `/etc/salt-api/pki/`? or something within `/etc/metalk8s`?

Reply:
Good catch @nootal. This key/cert should indeed go in `/etc/salt/pki/api` (one could argue for `/etc/salt/pki/master` somewhere as well, but there's a reason not to, read on). This way, there's no need to expose anything SaltAPI *doesn't* need (e.g., access to K8s API server certs, Salt master certs,...) to its container.

Reply:
Good point indeed.

That being said, I still need to mount the K8S stuff, to access the root CA, no?

Reply:
Done in a new commit.